subsys/net/lib/dns/resolve.c

```diff
@@ -316,21 +316,22 @@ static int dispatcher_cb(struct dns_socket_dispatcher *my_ctx, int sock,
 				continue;
 			}
 
+			ntry++;
 			ret = dns_write(ctx, j, i, dns_data->data, len,
 					net_buf_max_len(dns_data),
 					dns_cname, 0);
 			if (ret < 0) {
-				failure++;
+				nfail++;
 			}
 		}
 
-		if (failure) {
-			NET_DBG("DNS cname query failed %d times", failure);
-
-			if (failure == j) {
-				ret = DNS_EAI_SYSTEM;
-				goto quit;
-			}
+		if (nfail) {
```

Comment:
No implicit boolean conversions

```suggestion
		if (nfail > 0) {
```